### PR TITLE
More collection sorting 175

### PR DESCRIPTION
**Pull request by mccalluc**
@afred? Towards #175... though it doesn't get as far as sorting: it just does more title cleaning.

### CODE REVIEW

**Comment by mccalluc** [comment]
.... passing locally: On travis, not all the fixtures seem to be ingested. Looking into this...

**Comment by mccalluc** [comment]
Oh, I bet the class is being initialized before solr is populated...

**Comment by afred** [comment]
@mccalluc whoops.. sorry ignore comment. I was looking at an intermediate commit i think

File: app/models/tabbed.rb [hunk 1]
```diff
@@ -46,36 +46,61 @@ def norm(s)
       s.downcase.gsub(/\W+/, '-')
     end
 
+    def shared_prefix(titles)
+      tokens = titles.first.scan(/\S+\s+/)
+      prefixes = 0.upto(tokens.length).map { |i| tokens[0..i].join }.reverse
```
Comment (afred) [inline app/models/tabbed.rb]: @mccalluc may not be consequential here, but I don't think it should be `tokens.length - 1`

**Comment by afred** [comment]
@mccalluc this seems to be getting kinda hairy. I had to look at it for a while, but still not sure I grok it all.

Is the complexity due to the fact that we're trying to sort on something that is no longer a field by itself?

**Comment by mccalluc** [comment]
@afred: yeah, the complexity worries me a little, too. 
- Maybe we should just take the last part of the title? (ie, after the last semicolon)
- Or tolerate more redundancy on the collection pages?
- Or get FM to supply the short title that we want?

**Comment by afred** [comment]
@mccalluc do we not have the piece of metadata we need for sorting? I thought we were trying to sort by the interviewee's name (from looking at the tickets).

In my mind, sorting logic is a website concern, and we should be able to add easily sortable fields to the index as needed.

But if there's a case to push the sortability back to the data layer, I'd be ok with that too.

**Comment by mccalluc** [comment]
This example is interviewees, but the same logic ideally should be used on all the other collection pages, too.

**Comment by afred** [comment]
taking a step back, I think it would be much more straightforward to just have a field to reliably sort by.

We can already sort by the concatenated title. But we could also have a sortable `short_title`. I think that would simplify things in the long run.

And `short_title` field could be set like so..

```ruby
# in app/models/pb_core.rb

def interviewees
  contributors.select { |c| c.role == 'Interviewee' }
end

def to_solr
  # in the returned hash, add the following
  short_title: interviewees.empty? asset_title : interviewees[0].name
end
```

... or something like that.

**Comment by mccalluc** [comment]
Pending feedback from @Muraszko on #216: Tricky bit is that for some collections (New Television / Say Brother) we might want more distinguishing information than on others. If we can come up with logic that doesn't need to know the context, great.

**Comment by mccalluc** [comment]
@afred : After talking with Mike, we don't need any of this. yay?!

**Pull request closed without merging**